core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -223,6 +224,46 @@ class TransactionStateManager(brokerId: Int,
       throw new IllegalStateException(s"Unexpected empty transaction metadata returned while putting $txnMetadata")))
   }
 
+  def listTransactionStates(
+    filterProducerIds: Set[Long],
+    filterStateNames: Set[String]
+  ): Either[Errors, List[ListTransactionsResponseData.TransactionState]] = {
+    inReadLock(stateLock) {
+      if (loadingPartitions.nonEmpty) {
```

Comment:
Maybe we can return error only if the expected partition ids are included by loading partitions?

Reply:
The problem is that we don't have a way to map from the producerId to the partition.